### PR TITLE
Master mdl 71639 85

### PR DESCRIPTION
**Pull request by safatshahin**
PR for #85

### CODE REVIEW

File: question/bank/exporttoxml/lang/en/qbank_exporttoxml.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * @package    qbank_exporttoxml
+ * @copyright  2021 Catalyst IT Australia Pty Ltd
+ * @author     Safat Shahin <safatshahin@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+$string['pluginname'] = 'Export to XML';
+$string['privacy:metadata'] = 'Export to xml exports single question to xml, it does not store any user data.';
+
```
Comment (guillogo) [inline question/bank/exporttoxml/lang/en/qbank_exporttoxml.php:28]: please remove double new line.

File: question/bank/exporttoxml/classes/exporttoxml_helper.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ */
+class exporttoxml_helper {
+
+    /**
+     * Get the URL to export a single question (exportone.php).
+     *
+     * @param \stdClass|\question_definition $question the question definition as obtained from
+     *      question_bank::load_question_data() or question_bank::make_question().
+     *      (Only ->id and ->contextid are used.)
+     * @return \moodle_url the requested URL.
```
Comment (guillogo) [inline question/bank/exporttoxml/classes/exporttoxml_helper.php:46]: can you add please the doc for throwing exception `* @throws \moodle_exception` as moodle_url throws that one.

File: question/bank/exporttoxml/classes/exporttoxml_helper.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+     */
+    public static function question_get_export_single_question_url($question) {
```
Comment (guillogo) [inline question/bank/exporttoxml/classes/exporttoxml_helper.php:48]: can you please validate if is possible to add the return type here `: \moodle_url`

**Review by guillogo — CHANGES_REQUESTED** [review]
Hi @safatshahin 

The PR has been review with few and simple changes.

Code checker: OK
Unit Test: OK
User test: OK
backup, restore, import: OK
Code: few simple changes. 

Thank you.
Guillermo

File: question/bank/exporttoxml/classes/exporttoxml_helper.php (hunk 2 above)
Comment (safatshahin) [reply question/bank/exporttoxml/classes/exporttoxml_helper.php:46]: i was following the document from moodle and didnt use throws anywhere:
https://docs.moodle.org/dev/Setting_up_PhpStorm

which states:
Disable missing @throws tag warning in "Preferences / Project Settings / Inspections / PHP / PHPDoc"

i was assuming throws tag is something not recommended, any thoughts?

**Comment by safatshahin** [comment]
Hi @guillogo 
Thank you for reviewing this one. I have made some changes, just except one, made a comment there, please let me know what do you think.
Thanks

File: question/bank/exporttoxml/classes/exporttoxml_helper.php (hunk 2 above)
Comment (guillogo) [reply question/bank/exporttoxml/classes/exporttoxml_helper.php:46]: Hi @safatshahin some recent core things that use the @throws tag have been added. But, if core documentation says that is not mandatory, then is fine. Thank you.

**Review by guillogo — APPROVED** [review]
Approved.

**Comment by guillogo** [comment]
Hi @safatshahin approved and merged. Thank you.

**Pull request merged**